framework/common/encryption.cpp

```diff
@@ -347,16 +347,30 @@ bool wps_calculate_keys(const diffie_hellman &dh, const uint8_t *remote_pubkey,
     return true;
 }
 
-bool kwa_compute(const uint8_t *authkey, uint8_t *data, uint32_t data_len, uint8_t *kwa)
+bool kwa_compute(const uint8_t *authkey, uint8_t *data, uint32_t data_len, uint8_t *kwa, bool more)
 {
     uint8_t hmac_[32];
-    hmac hmac_kwa(authkey, 32);
-    if (!hmac_kwa.update(data, data_len))
-        return false;
-    if (!hmac_kwa.digest(hmac_))
-        return false;
+    static hmac *hmac_kwa = nullptr;
```

Comment:
Hm, I really don't like it when functions are made non-reentrant... I much prefer the copy in that case.

Basically, we should use the hmac class directly instead of kwa_compute.

Or keep it as it was :-)

Reply:
we'll kill the fixup